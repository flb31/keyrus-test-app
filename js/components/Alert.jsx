import React, { Component } from 'react';
import { Alert as BootstrapAlert } from 'react-bootstrap';
import PropTypes from 'prop-types';

export const typesAlert = { success: 'success', warning: 'warning', danger: 'danger', info: 'info' };

class Alert extends Component {
  state = {
    showAlert: true
  };

  handleDismiss = () => {
    this.setState({ showAlert: false });
  };

  render() {
    const { alert } = this.props;
    
    if(!this.state.showAlert) {
        return null;
    }

    return <BootstrapAlert bsStyle={alert.type} onDismiss={this.handleDismiss}>{alert.message}</BootstrapAlert>;
  }
}

Alert.propTypes = {
  alert: PropTypes.shape({
    type: PropTypes.string,
    message: PropTypes.string
  }).isRequired
};

export default Alert;
