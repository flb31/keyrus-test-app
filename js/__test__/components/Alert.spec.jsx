import React from 'react';
import { shallow } from 'enzyme';
import Alert from '../../components/Alert';

const alert = {
  id: '123456',
  type: 'success',
  message: 'Message okay'
};

test('Alert renders correctly', () => {
  const component = shallow(<Alert alert={alert} handleAlert={() => {}} />);
  expect(component).toMatchSnapshot();
});
