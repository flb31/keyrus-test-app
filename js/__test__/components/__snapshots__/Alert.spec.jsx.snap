// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<Alert
  bsClass="alert"
  bsStyle="success"
  className="Alert Alert__success"
  closeLabel="Close alert"
  onDismiss={[Function]}
>
  Message okay
</Alert>
`;
